.github/workflows/pre-merge.yml

```diff
@@ -16,10 +16,12 @@ jobs:
         uses: actions/setup-node@v4
         with:
           node-version: "20"
-          cache: "yarn"
+
+      - name: Enable Corepack
+        run: corepack enable
 
       - name: Install dependencies
-        run: yarn install --frozen-lockfile --no-progress --non-interactive
+        run: yarn install --immutable
 
```

Comment:
We need to add `yarn dedupe` lock check manually now, after removal of `postinstall` script. For that we should use [`check` flag](https://yarnpkg.com/cli/dedupe#details) which errors out when lock change is encountered:

```sh
yarn dedupe --check
```